// @stafull/shared/src/components/organisms/ProtectedRoute/ProtectedRoute.jsx
/**
 * ProtectedRoute - Route wrapper for authenticated routes
 *
 * Features:
 * - Redirects to auth portal if not authenticated
 * - Role-based access control
 * - Loading state while checking auth
 */

import { useAuth } from '../../../hooks/useAuth';

const AUTH_URL = import.meta.env.DEV ? 'http://localhost:5170' : 'https://auth.stafull.com';

export function ProtectedRoute({
  children,
  roles,
  fallback = null,
  redirectToAuth = true,
}) {
  const { user, loading, isAuthenticated } = useAuth();

  // Show loading state
  if (loading) {
    return fallback || (
      <div style={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        minHeight: '100vh',
        color: 'var(--color-text-secondary, #6b7280)',
      }}>
        Loading...
      </div>
    );
  }

  // Not authenticated
  if (!isAuthenticated) {
    if (redirectToAuth) {
      // Redirect to auth portal with return URL
      const returnUrl = encodeURIComponent(window.location.href);
      window.location.href = `${AUTH_URL}?return=${returnUrl}`;
      return null;
    }
    return fallback;
  }

  // Check role access
  if (roles && roles.length > 0) {
    const hasAccess = roles.includes(user.role);

    if (!hasAccess) {
      // User doesn't have required role
      return (
        <div style={{
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'center',
          alignItems: 'center',
          minHeight: '100vh',
          gap: 'var(--space-4, 18px)',
          color: 'var(--color-text-secondary, #6b7280)',
        }}>
          <h1 style={{ color: 'var(--color-text-primary, #111827)' }}>
            Access Denied
          </h1>
          <p>You don&apos;t have permission to view this page.</p>
          <p style={{ fontSize: '14px' }}>
            Required role: {roles.join(' or ')}
          </p>
        </div>
      );
    }
  }

  // User is authenticated and has required role
  return children;
}

export default ProtectedRoute;
