# BUILD RULES
## StaFull Development Standards - Mandatory Reference
**Version:** 3.2 | **Updated:** 2026-01-24 | **Status:** ENFORCED

> **DESIGN AUTHORITY:** `stafull-design-system/src/stafull-system.css` (HIGHEST)
> **DESIGN RULES:** See `DESIGN-RULES.md` for all visual design specifications
> **MANDATORY:** All agents and developers MUST read this file before ANY code changes.

---

## 0. DESIGN SYSTEM REFERENCE (Source: stafull-design-system)

**This file covers BUILD rules only. For visual design specifications, see DESIGN-RULES.md.**

### Critical Design Tokens (DO NOT CONTRADICT)
| Token | Value | Usage |
|-------|-------|-------|
| Card radius | 14px | `--sf-radius-card` |
| Button radius | 8px | `--sf-radius-button` |
| Card border | `1px solid rgba(255, 255, 255, 0.04)` | All cards |
| Focus state | `outline: 1px solid #a30000; outline-offset: 0` | ALL focus |
| Transitions | 100ms, 200ms, 300ms ONLY | No other values |
| Shadows | BLACK ONLY | `rgba(0,0,0,...)` - no colored shadows |
| Font weights | 400, 500, 600, 700 ONLY | No 300, 800, 900 |

---

## 1. CORE PRINCIPLES

### No Band-Aid Fixes
- **ROOT-LEVEL changes ONLY**
- Never patch symptoms - fix causes
- No temporary workarounds
- No `!important` overrides (fix the root CSS)

### Delete Temp Code
- Remove ALL workarounds after proper fixes
- No commented-out old implementations
- No deprecated code left in codebase
- No `// TODO: remove later` that stays forever

### Single Source of Truth
- One authoritative source per component
- Design tokens in `stafull-shared/src/tokens/tokens.css`
- Rules in `stafull-shared/docs/rules/`
- No duplicate token definitions

---

## 2. CODE STANDARDS

### CSS Variables
```css
/* CORRECT - use tokens */
padding: var(--space-4);
color: var(--red);
box-shadow: var(--shadow-md);

/* WRONG - hardcoded values */
padding: 14px;
color: #a30000;
box-shadow: 0 4px 6px rgba(0,0,0,0.15);
```

### Console Logging
```javascript
// CORRECT - wrapped in DEV check
if (import.meta.env.DEV) console.log('[PREFIX] message', data);

// WRONG - bare console.log
console.log('debug', data);
```

### Import Organization
```javascript
// 1. External packages
import { useState, useEffect } from 'react';
import { Routes, Route } from 'react-router-dom';

// 2. Internal absolute imports
import { Button, Card } from '@stafull/shared/components';

// 3. Relative imports
import styles from './Component.module.css';
```

---

## 3. AUTHENTICATION

### Login Components
- **NO `<form>` tags** in login components
- Use `<div>` with onClick handlers
- Prevents default form submission issues
- Unified login routing by account type

```jsx
// CORRECT
<div className={styles.loginContainer}>
  <input type="email" ... />
  <input type="password" ... />
  <button onClick={handleLogin}>Login</button>
</div>

// WRONG
<form onSubmit={handleLogin}>
  ...
</form>
```

---

## 4. DASHBOARD PATTERN

**Dashboard = Menu** - Core design principle.

### Rules
- NO sidebars on main dashboard view
- NO text-based navigation menus
- Dashboard is ENTIRELY clickable department cards
- Cards show: icon, title, description/metrics
- Card click → enter department or open modal
- Logo click → ALWAYS returns to dashboard

### Dashboard Card Counts
| Portal | Cards | Dashboard Name |
|--------|-------|----------------|
| Holdings Admin | 10 | Command Center |
| Franchise Owner | 8 | P&L Cockpit |
| Manager | 20 | Mission Control |
| Customer | 7 | - |
| Employer | 7 | - |
| Employee | 7 | - |
| Investor | 6 | READ-ONLY |
| SBA Lender | 6 | READ-ONLY |
| Personnel | 9 | - |
| Driver | 4 modes | Co-Pilot |

### Responsive Grid
| Breakpoint | Cards per Row |
|------------|---------------|
| Desktop (1280px+) | 4 cards |
| Tablet (768-1279px) | 2 cards |
| Mobile (<768px) | 1 card |

### FranchiseOS Dashboard Components

Use shared components from `@stafull/shared/components/molecules`:

**DashboardCard** - Individual metric card
```jsx
import { DashboardCard } from '@stafull/shared/components/molecules';

<DashboardCard
  title="NET PROFIT"
  icon={ICON_PATHS.finance}
  primaryMetric="$47,820"
  metricLabel="This Month"
  trend="+8.2%"
  trendDirection="up"
  size="large"        // large, tall, medium, wide, small
  onClick={() => navigate('/finance')}
/>
```

**DashboardRow** - Row layout container
```jsx
import { DashboardRow } from '@stafull/shared/components/molecules';

// Row with 2/3 + 1/3 layout
<DashboardRow layout="large-small">
  <DashboardCard title="MAIN" size="large" />
  <DashboardCard title="SIDE" size="tall" />
</DashboardRow>

// Row layouts: equal, large-small, small-large, half, thirds, quarters
```

**DashboardHeader** - Header with logo, title, settings
```jsx
import { DashboardHeader } from '@stafull/shared/components/molecules';

<DashboardHeader
  title="Network Command Center"
  logoSrc="/logos/stafull-logo-dark.svg"
  onLogoClick={() => navigate('/')}
  onSettingsClick={() => openSettings()}
/>
```

### CSS Module Pattern for Dashboards

When building custom dashboard layouts, use CSS Modules:

```css
/* Dashboard.module.css */
.dashboard {
  display: flex;
  flex-direction: column;
  gap: var(--space-5);
  max-width: 1400px;
  margin: 0 auto;
  padding: var(--space-5);
}

.row {
  display: grid;
  gap: var(--space-5);
}

/* Row layout: 2/3 + 1/3 */
.row:nth-child(1) {
  grid-template-columns: 2fr 1fr;
}

/* Row layout: thirds */
.row:nth-child(2) {
  grid-template-columns: 1fr 1fr 1fr;
}

/* Responsive: tablet */
@media (max-width: 1024px) {
  .row:nth-child(1),
  .row:nth-child(2) {
    grid-template-columns: 1fr;
  }
}
```

---

## 5. DRIVER PORTAL

### Orientation
- **Landscape tablet ONLY** - No portrait mode
- **NO scrolling** - Use modals for overflow
- Portrait shows "Please rotate device" overlay

### 4 Operational Modes
| Mode | Purpose |
|------|---------|
| Parked | Shift start/end, vehicle status |
| Driving | Navigation map (75%), info strip (25%) |
| Delivery | Customer info, fuel type, confirmation |
| Resupply | Tank levels, fill confirmation |

### Layout
- DrivingMode: 75% map (left), 25% info strip (right)
- Large touch targets: 48-80px minimum
- Limited buttons for single-finger use

### AI Cameras
- **NO manual photo buttons**
- AI cameras handle all photos automatically
- AI supervised and verified
- DriverIncidentReport uses AI camera data

---

## 6. GIT WORKFLOW

### Commit Messages
```
feat: add new feature
fix: resolve bug
docs: update documentation
refactor: restructure code
style: formatting changes
test: add tests
chore: maintenance
```

### Branch Naming
- `main` - production branch
- `feature/feature-name` - new features
- `fix/bug-description` - bug fixes

### Commit Rules
- Reference ticket/issue when applicable
- Keep commits atomic and focused
- Never commit secrets or credentials

---

## 7. PROCESS MANAGEMENT

### NEVER Use
```bash
# BANNED - kills Claude Code
taskkill /F /IM node.exe
```

### Use Instead
```bash
# Kill stuck servers by port
npx kill-port 3001
npx kill-port 5174

# Or use Ctrl+C in terminal
```

---

## 8. TEST CREDENTIALS

**Password for all test accounts:** `pass`

| Role | Email |
|------|-------|
| Holdings Admin | admin@test.com |
| Franchise Owner | owner@test.com |
| Manager | manager@test.com |
| Driver | driver@test.com |
| Customer | customer@test.com |
| Employer Admin | employer@test.com |
| Employee | employee@test.com |
| Investor | investor@test.com |
| SBA Lender | lender@test.com |

**9 demo roles only.** Fleet Manager and Dispatcher removed (merged into Management).

---

## 9. FILE ORGANIZATION

### Component Structure
```
ComponentName/
├── ComponentName.jsx
├── ComponentName.module.css
├── ComponentName.test.jsx (if tests)
└── index.js (re-export)
```

### CSS Module Naming
```css
/* Use camelCase for class names */
.container { }
.cardTitle { }
.buttonPrimary { }

/* WRONG - kebab-case */
.card-title { }
.button-primary { }
```

---

## 10. HR MODULE (People Operations)

### Structure
HR Module is a **standalone module** accessible from:
- Franchise Owner Portal
- Management Portal
- Employee self-service

### Automation Level: 90%

### Employer-Facing Features
| Feature | Description |
|---------|-------------|
| Workforce Overview | Total employees, attrition rate |
| Payroll Preview | Next payroll, bonus pool |
| Training & Development | Compliance rates, assigned tasks |
| Performance Management | Reviews due, team scores |
| Recruitment Pipeline | Open jobs, applicants |
| Compliance Center | Next audit date |

### Employee-Facing Features
| Feature | Description |
|---------|-------------|
| Payslip | Current net pay, download PDF |
| Schedule | Upcoming shifts, swap requests |
| Benefits Summary | Plan, coverage |
| Performance Score | Current score display |

### Automated Workflows
- Resume screening with AI scoring
- Background check automation
- Interview scheduling
- Offer letter generation
- Electronic document collection
- Training schedule automation
- Payroll processing integration

---

## 11. FINANCE MODULE

### Structure
Finance Module is a **standalone module** with automation level: 95%

### Transaction Processing
- Bank feed integration with auto-categorization
- Expense receipt capture and processing
- Invoice generation and tracking
- Payment processing automation
- Reconciliation automation

### Financial Reporting
- Real-time P&L statement generation
- Balance sheet automation
- Cash flow statement generation
- Financial ratio calculation
- Performance benchmarking

### Subscription Billing
- Tiered pricing management (weekly/bi-weekly/annual)
- Automated billing and payment collection
- Failed payment handling
- Refund processing automation
- Revenue recognition automation

### Expense Management
- Fuel cost management with card-lock monitoring
- Variable cost tracking and analysis
- Budget management tools
- Expense approval workflow automation
- Vendor management system

### Tax & Compliance
- Automated tax calculation
- Electronic filing system
- Tax payment scheduling
- Compliance reporting automation
- Fraud detection algorithms

---

## 12. PROHIBITED PATTERNS

```javascript
// BANNED - direct DOM manipulation
document.getElementById('...');
document.querySelector('...');

// BANNED - inline styles with hardcoded values
style={{ padding: '14px' }}

// BANNED - class name strings (use CSS modules)
className="card-container"

// CORRECT - CSS modules
className={styles.cardContainer}
```

---

**THIS FILE IS ENFORCED. NO EXCEPTIONS. ROOT-LEVEL FIXES ONLY.**
