# DECISIONS LOG
## All Finalized Decisions with Timestamps
**Version:** 3.3 | **Updated:** 2026-01-25 | **Status:** ACTIVE

> **PURPOSE:** Track all finalized decisions. When approved changes are made, log them here.

---

## HOW TO UPDATE THIS FILE

When a new decision is finalized:
1. Add entry below with date/time
2. Category: Design | Build | Architecture | Business
3. Clear decision statement
4. Update relevant rules file if applicable
5. Commit with message: `docs: log decision - [title]`

---

## DESIGN DECISIONS

### 2024-12-XX: Red Range Finalized
- **Category:** Design
- **Decision:** Red values locked to #a30000 (primary), #cc0000 (bright), #8a0000 (dark)
- **Gradient:** `linear-gradient(160deg, #cc0000, #a30000, #8a0000)`
- **Rule:** Red is glossy/reflective ONLY, never glowing in shadows

### 2024-12-XX: Typography Finalized
- **Category:** Design
- **Decision:** Montserrat font ONLY, no other fonts allowed
- **Weights:** 400, 500, 600, 700 ONLY
- **Banned:** 300, 800, 900

### 2024-12-XX: Spacing System Finalized
- **Category:** Design
- **Decision:** Only these spacing values: 4, 6, 10, 14, 18, 30, 45, 90 px
- **Source:** Donald's original scale

### 2024-12-XX: Animation Durations Finalized
- **Category:** Design
- **Decision:** Only these durations: 100ms, 200ms, 300ms

### 2024-12-XX: Focus Rings Finalized
- **Category:** Design
- **Decision:** 1px width ONLY (not 2-3px), color #a30000, no glow

### 2024-12-XX: Icons Rule
- **Category:** Design
- **Decision:** SVG icons only, no emojis in UI

### 2024-12-XX: Shadow Rule
- **Category:** Design
- **Decision:** Black/neutral shadows ONLY, NO red in shadows ever

### 2026-01-16: Font Weight Restriction
- **Category:** Design
- **Decision:** Font weights 300, 800, 900 are BANNED
- **Allowed:** 400 (regular), 500 (medium), 600 (semibold), 700 (bold)
- **Files Updated:** DESIGN-RULES.md, tokens.css (already compliant)

### 2026-01-16: Logo Source Changed
- **Category:** Design
- **Decision:** Logo source changed from CSS to SVG files
- **Old:** `/assets/css/stafull-logo.css` (DELETED)
- **New:** `stafull-shared/assets/logos/*.svg` (12 files)
- **Files Updated:** DESIGN-RULES.md

### 2026-01-16: FranchiseOS Dashboard Layout System (APPROVED)
- **Category:** Design
- **Decision:** Adopt FranchiseOS layout philosophy for dashboards
- **Details:**
  - Row-based layouts with variable card sizes (Large 2/3, Tall 1/3, Medium 1/2, Wide 2/3, Small 1/3)
  - Header: Logo (left), Dashboard title (center), Settings (right)
  - Cards are clickable menus with primary metrics + trend indicators
  - Asymmetrical layouts based on content importance
- **Files Updated:** DESIGN-RULES.md (Section 10)

### 2026-01-16: Data Visualization Standards (APPROVED)
- **Category:** Design
- **Decision:** Adopt FranchiseOS chart/graph standards
- **Details:**
  - Bars: Solid fill, soft rounded corners
  - Lines: Thin, smooth curves
  - Donuts: Thin strokes, minimal labeling
  - Primary metrics: Large bold (40px+) with trend arrows
- **Files Updated:** DESIGN-RULES.md (Section 11)

### 2026-01-16: Business Data Specifications (APPROVED)
- **Category:** Design
- **Decision:** Document subscription tiers and telemetry intervals
- **Details:**
  - Subscription tiers: Weekly, Bi-weekly, Annual
  - Driver compensation: 80% hourly + 0-40% bonus
  - GPS telemetry: 5-second intervals
- **Files Updated:** DESIGN-RULES.md (Section 12)

### 2026-01-16: Dashboard Redesign with Shared Components
- **Category:** Design
- **Decision:** Implement FranchiseOS dashboard components as shared reusable modules
- **Details:**
  - Created DashboardCard component with metrics, trends, progress bars
  - Created DashboardHeader component with logo/title/settings layout
  - Created DashboardRow component with 6 layout options
  - Card sizes: large (280px), tall (280px), medium (180px), wide (160px), small (140px)
  - Dark theme with --gray-850 backgrounds, --gray-900 page background
  - Primary metrics 40px bold white, card titles 10px uppercase
- **Components Created:**
  - `stafull-shared/src/components/molecules/DashboardCard/`
  - `stafull-shared/src/components/molecules/DashboardHeader/`
  - `stafull-shared/src/components/molecules/DashboardRow/`
- **Files Updated:** DESIGN-RULES.md (Section 10, 10.1, 10.2), BUILD-RULES.md (Section 4)

### 2026-01-25: Design File Conflict Resolution
- **Category:** Design
- **Decision:** Resolved critical conflicts between design system source files
- **Details:**
  - Fixed duplicate @keyframes shimmer: renamed skeleton animation to `shimmer-skeleton`
  - Fixed .sf-card::before conflict: aligned components.css to use 50% height sheen (matches tokens.css)
  - Added missing radius variables: `--radius-base` (8px), `--radius-badge` (50px)
  - Synced shadow tokens: added --sf-shadow-card and --sf-shadow-card-hover to tokens.css
- **Files Updated:** stafull-system.css, tokens.css, components.css

### 2026-01-25: Portal Rule Violations Fixed
- **Category:** Build
- **Decision:** Fixed high-priority rule violations across all portals
- **Details:**
  - Removed red shadow in portal-driver (DrivingMode.module.css)
  - Removed all !important declarations (portal-hq, portal-franchise)
  - Fixed non-standard transitions: 500ms→300ms, 150ms→200ms
  - Used CSS specificity doubling instead of !important for fullscreen map
- **Files Updated:**
  - portal-driver: DrivingMode.module.css
  - portal-hq: Dashboard.module.css
  - portal-franchise: Finance.module.css, Routes.module.css, OwnerDashboard.module.css
  - portal-customer: B2CDashboard.css, ResidentialDashboard.css

### 2026-01-24: Design System Audit - Button Animations
- **Category:** Design
- **Decision:** Standardized button shimmer and loading spinner patterns
- **Details:**
  - Shimmer animation on `::before` pseudo-element (hover only, 1.2s duration)
  - Loading spinner on `::after` pseudo-element (centered with margin offsets)
  - Disabled buttons hide shimmer (`::before { display: none; }`)
  - Loading state hides shimmer to prevent animation conflicts
- **Files Updated:** stafull-system.css, components.css, tokens.css

### 2026-01-24: Light Theme Shadows - Heavier Visibility
- **Category:** Design
- **Decision:** Increased light theme shadow opacity for better visibility
- **Details:**
  - `--shadow-card-light: 0 2px 6px rgba(0,0,0,0.18), 0 10px 24px rgba(0,0,0,0.15)`
  - `--shadow-card-hover-light: 0 4px 10px rgba(0,0,0,0.22), 0 20px 40px rgba(0,0,0,0.2)`
  - `--shadow-lg-light: 0 2px 4px rgba(0,0,0,0.15)`
- **Files Updated:** stafull-system.css, tokens.css

### 2026-01-24: Status Colors Finalized
- **Category:** Design
- **Decision:** Status colors locked to pure/saturated values
- **Details:**
  - Success: #00ff00 (Pure Green)
  - Warning: #eab308 (Yellow-Orange)
  - Error: #ff4500 (Red-Orange)
  - Info: #0000ff (Pure Blue)
- **Files Updated:** stafull-system.css, tokens.css

### 2026-01-24: Light Theme Border Colors
- **Category:** Design
- **Decision:** All light theme borders must use gray-400 for consistency
- **Details:**
  - Updated all `border-color: var(--gray-300)` to `var(--gray-400)` in light theme contexts
  - Applies to: buttons, spinners, modal close, cards
- **Files Updated:** stafull-system.css

### 2026-01-24: Fuel Colors Synced
- **Category:** Design
- **Decision:** Fuel colors synced across all source files
- **Details:**
  - Diesel: #00ff00 (Pure Green)
  - Regular: #eab308 (Yellow)
  - Premium: #ff4500 (Orange-Red)
  - DEF: #0000ff (Pure Blue)
- **Files Updated:** stafull-system.css (was using old values)

### 2026-01-16: HQ Dashboard Redesigned
- **Category:** Design
- **Decision:** Redesigned HQ Dashboard as "Network Command Center" with FranchiseOS layout
- **Details:**
  - Row 1: Franchise Network map (2/3) + Network Health (1/3)
  - Row 2: AI Engine | Route Optimization | Safety Score (thirds)
  - Row 3: Financial Summary | Marketing & Leads (halves)
  - Row 4: Franchise Sales | Compliance | SBA (thirds)
  - 10 cards total with mock data for demo mode
- **Files Updated:** stafull-portal-hq/src/pages/Dashboard.jsx, Dashboard.module.css

### 2026-01-16: Owner Dashboard Redesigned
- **Category:** Design
- **Decision:** Redesigned Owner Dashboard as "P&L Cockpit" with financial focus
- **Details:**
  - Row 1: Net Profit with P&L breakdown + trend chart (2/3) | Health Score donut (1/3)
  - Row 2: HQ Fees breakdown | Operations metrics (halves)
  - Row 3: Team | Fleet | Subscribers (thirds)
  - Row 4: Reports & Analytics links (full width)
  - 8 cards total with detailed financial metrics
- **Files Updated:** stafull-portal-franchise/src/pages/OwnerDashboard.jsx, OwnerDashboard.module.css

### 2026-01-16: Management Dashboard Redesigned
- **Category:** Design
- **Decision:** Redesigned Management Dashboard as "Mission Control" with sectioned layout
- **Details:**
  - Dispatch Section: Live map, deliveries, routes, assignments
  - Team & Finance Section: driver status, scheduling, payroll, training, P&L, invoices, expenses
  - Safety & Fleet Section: safety score, incidents, inspections, vehicles, tanks, maintenance
  - Customers & Reports Section: subscribers, satisfaction, reports
  - 20 cards total organized by functional category
- **Files Updated:** stafull-portal-franchise/src/pages/ManagementDashboard.jsx, ManagementDashboard.module.css

---

## BUILD DECISIONS

### 2024-12-XX: No Band-Aid Fixes
- **Category:** Build
- **Decision:** All changes must be root-level, no patching symptoms

### 2024-12-XX: Dashboard = Menu Pattern
- **Category:** Build
- **Decision:** Portal dashboards ARE navigation menus with clickable cards, no traditional sidebars

### 2024-12-XX: Authentication Pattern
- **Category:** Build
- **Decision:** No `<form>` tags in login components, use `<div>` with onClick handlers

### 2024-12-XX: Driver Portal Requirements
- **Category:** Build
- **Decision:** Landscape tablet ONLY, no portrait, no scrolling, 4 modes, AI cameras automatic

### 2024-12-XX: Demo Login Roles
- **Category:** Build
- **Decision:** 9 roles only
- **Included:** Holdings Admin, Franchise Owner, Manager, Driver, Customer, Employer Admin, Employee, Investor, SBA Lender
- **Excluded:** Fleet Manager, Dispatcher (merged into Management)

### 2026-01-16: Driver Portal Complete
- **Category:** Build
- **Decision:** Driver Portal fully implemented with 6 components
- **Components:** Dashboard.jsx, Settings.jsx, ParkedMode.jsx, DrivingMode.jsx, DeliveryMode.jsx, ResupplyMode.jsx
- **Compliance:** All MASTER-RULES followed

### 2026-01-16: HR Module as Standalone (APPROVED from FranchiseOS)
- **Category:** Build
- **Decision:** HR Module ("People Operations") is a standalone module, NOT part of shared/personnel
- **Details:**
  - 90% automation level
  - Accessible from: Franchise Owner Portal, Management Portal, Employee self-service
  - Features: Workforce overview, payroll preview, training, performance, recruitment, compliance
- **Files Updated:** BUILD-RULES.md (Section 10)

### 2026-01-16: Finance Module as Standalone (APPROVED from FranchiseOS)
- **Category:** Build
- **Decision:** Finance Module is a standalone module with 95% automation
- **Details:**
  - Transaction processing with auto-categorization
  - Subscription billing (weekly/bi-weekly/annual)
  - Expense management with card-lock monitoring
  - Tax & compliance automation
- **Files Updated:** BUILD-RULES.md (Section 11)

---

## ARCHITECTURE DECISIONS

### 2025-01-15: Repository Structure
- **Category:** Architecture
- **Decision:** 6 repos - stafull-shared (source of truth), core-engine, portal-hq, portal-franchise, portal-driver, portal-customer

### 2025-01-15: Master Reference Consolidation
- **Category:** Architecture
- **Decision:** All design rules consolidated into stafull-shared/docs/rules/

### 2026-01-16: Port Standardization
- **Category:** Architecture
- **Decision:** Fixed port assignments
- **Ports:** core-engine=3001, hq=5174, franchise=5175, driver=5176, customer=5177

### 2026-01-16: Documentation Consolidation
- **Category:** Architecture
- **Decision:** Consolidated all rules into categorized files in stafull-shared/docs/rules/
- **Files:** DESIGN-RULES.md, BUILD-RULES.md, ARCHITECTURE-RULES.md, DECISIONS-LOG.md
- **Removed:** MASTER-RULES.md, duplicate docs

### 2026-01-16: Transition Token Standardization
- **Category:** Architecture
- **Decision:** Renamed `--transition-normal` to `--transition-base` everywhere
- **Standard:** --transition-fast (100ms), --transition-base (200ms), --transition-slow (300ms)

### 2026-01-16: FranchiseOS Module Architecture (APPROVED)
- **Category:** Architecture
- **Decision:** Adopt FranchiseOS module structure
- **Details:**
  - Core AI Engine: 99.9% automated dispatch, route optimization, compliance
  - HQ Portal: Network Command Center
  - Franchisee Owner Portal: P&L Cockpit
  - Management Portal: Mission Control
  - Driver Tablet App: Co-Pilot with 4 modes
  - Customer Portal: Account, delivery, savings management
  - HR Module: 90% automated, standalone
  - Finance Module: 95% automated, standalone
- **Files Updated:** ARCHITECTURE-RULES.md (Section 1)

---

## BUSINESS DECISIONS

### 2024-12-XX: Logo Implementation
- **Category:** Business
- **Decision:** Single source of truth at `stafull-shared/assets/logos/*.svg`
- **Details:** Capital S and F, macron over "a" only, macron specs: top 0.145em, left 17.5%, width 65%, height 0.09em

### 2024-12-XX: Vehicle Configuration
- **Category:** Business
- **Decision:** 2026 BrightDrop 400 2WD, 8-tank config (3×100gal gas, 4×100gal diesel, 1×100gal DEF), 670 gal total

### 2024-12-XX: FMCSA Compliance
- **Category:** Business
- **Decision:** Diesel in non-bulk containers (<119 gal) = combustible, no hazmat regs, no CDL endorsement, no placards

---

## TEMPLATE FOR NEW DECISIONS

```markdown
### YYYY-MM-DD: [Decision Title]
- **Category:** Design | Build | Architecture | Business
- **Decision:** [Clear statement of what was decided]
- **Details:** [Additional context if needed]
- **Files Updated:** [List of files that need updating]
```

---

**LOG ALL APPROVED CHANGES HERE.**
