# StaFull Design System v2.0
## Authoritative Design Rules

> **Status:** FINALIZED
> **Created:** 2026-01-17
> **Archived v1:** _ARCHIVED_DESIGN-RULES_v1.md
> **Authority:** Donald B. Havery

---

# SECTION 0: BRAND & NAMING (HARD LOCK)

## 0.1 Approved Brand Forms (ONLY THESE)
- **StƒÅFull**
- **StƒÅFull.com**
- **StƒÅFull, LLC**
- **StƒÅFull Holdings, LLC**

## 0.2 Explicitly Forbidden
- ‚ùå ALL CAPS (STAFULL, STƒÄFULL)
- ‚ùå Incorrect capitalization (Stafull, staFull)
- ‚ùå Umlauts (St√§Full)
- ‚ùå Circumflex or other diacritics (St√¢Full)
- ‚ùå Any brand text not listed above
- ‚ùå Removing macron unless technically impossible

**If any invalid brand form appears ‚Üí the output is WRONG.**

---

# SECTION 1: CORE PHILOSOPHY (NON-NEGOTIABLE)

## 1.1 Design Intent
This system must look like:
- A **premium industrial control interface**
- A **physical surface**, not a flat web page
- A **dashboard composed of tangible objects**
- **Automotive / aerospace / enterprise-grade**

## 1.2 Automatic Failures
If anything looks:
- Web-default
- Flat
- Template-like
- SaaS-generic
- Like a typical admin dashboard

**‚Üí It is WRONG. Rebuild it.**

---

# SECTION 2: TARGET CANVAS

## 2.1 Default Viewport (Desktop 16:9)
| Property | Value |
|----------|-------|
| Primary Target | Desktop widescreen |
| Aspect Ratio | 16:9 |
| Width | 1440‚Äì1920px |
| Height | 900‚Äì1080px |
| Scrolling | **NONE** on primary dashboard - entire view must fit |

## 2.2 Responsive Breakpoints
| Breakpoint | Width | Layout | Scrolling |
|------------|-------|--------|-----------|
| Ultra-wide | 1920px+ | 3-column (300px / flex / 340px) | Natural |
| Desktop (default) | 1440‚Äì1919px | 3-column (320px / flex / 360px) | Natural |
| Tablet Landscape | 1024‚Äì1439px | 3-column (260px / flex / 300px) | Enabled |
| Tablet Portrait | 768‚Äì1023px | 2-column (left hidden) | Enabled |
| Mobile Landscape | 480‚Äì767px | 1-column | Enabled |
| Mobile Portrait | < 480px | 1-column compact | Enabled |
| Minimum | 320px | 1-column ultra-compact | Enabled |

## 2.3 Scrolling Behavior
- **Desktop (1440px+)**: Content flows naturally, scrollbar appears if window too short
- **Tablet and below**: Always scrollable, cards never compress or overlap
- **Cards must NEVER**: Layer over each other, shrink to unreadable sizes, or get cut off

## 2.4 Max-Width Constraint
- Content max-width: **1920px** (centered)
- Background extends infinitely beyond max-width
- Prevents over-stretching on ultra-wide monitors

## 2.5 DO
- Design primarily for desktop widescreen 16:9
- Layout must visually fill canvas horizontally AND vertically at desktop
- Use 3-column dashboard architecture at desktop
- Allow natural scrolling when content exceeds viewport
- Support responsive layouts for tablet/mobile

## 2.6 DO NOT
- ‚ùå Produce narrow, poster-like, or vertical layouts at desktop
- ‚ùå Leave unused vertical space below cards
- ‚ùå Center everything in a single column at desktop
- ‚ùå Hide content off-screen
- ‚ùå Force cards to compress/overlap to fit viewport
- ‚ùå Use fixed height constraints that clip content

---

# SECTION 3: LAYOUT SYSTEM ‚Äî 3-COLUMN DASHBOARD (MANDATORY)

## 3.1 Column Structure
| Column | Purpose | Desktop Width |
|--------|---------|---------------|
| **Left** | Design system / controls / navigation | 320px |
| **Center** | Primary operational content (map, route, orders) | flex (1fr) |
| **Right** | Status, alerts, metrics, leaderboards | 360px |

## 3.2 Grid CSS (Responsive)
```css
/* Page container */
.page {
  min-height: 100vh;
  overflow-y: auto;  /* Natural scrolling */
}

/* Main grid - Desktop default */
.main {
  display: grid;
  grid-template-columns: 320px 1fr 360px;
  gap: 12px;  /* --sf-gap: Source: stafull-design-system */
  padding: 16px;
  max-width: 1920px;
  margin: 0 auto;
}

/* Column */
.column {
  display: flex;
  flex-direction: column;
  gap: 12px;  /* --sf-gap */
}

/* Ultra-wide (1920px+) */
@media (min-width: 1920px) {
  .main { grid-template-columns: 300px 1fr 340px; }
}

/* Tablet Landscape (1024-1439px) */
@media (max-width: 1439px) {
  .main {
    grid-template-columns: 260px 1fr 300px;
    gap: 12px;
    padding: 14px 18px 24px;
  }
}

/* Tablet Portrait (768-1023px) */
@media (max-width: 1023px) {
  .main { grid-template-columns: 1fr 1fr; }
  .column:first-child { display: none; }  /* Hide left column */
}

/* Mobile (< 768px) */
@media (max-width: 767px) {
  .main { grid-template-columns: 1fr; }
}
```

## 3.3 DO
- Columns must be visually distinct but harmonious
- Center column is always widest
- Cards may span vertically but must remain dense
- Let cards align only when it looks intentional
- Allow cards to span more width, be taller/shorter based on content
- Keep gaps consistent (12px) - Source: stafull-design-system `--sf-gap`

## 3.4 DO NOT
- ‚ùå Force strict equal-height columns
- ‚ùå Float cards arbitrarily
- ‚ùå Allow cards to "almost align" ‚Äî either fully align or clearly offset
- ‚ùå Leave "dead air" between cards
- ‚ùå Leave big empty patches with no cards
- ‚ùå Create long, empty, stretched cards only to fill a column

---

# SECTION 4: BACKGROUND RULES

## 4.1 Background Philosophy
Background is a **SATIN** surface ‚Äî never flat, never matte.
**Background must be LIGHTER than cards** so cards float on top visually.

## 4.2 DO
- Use a single continuous background
- Use **warm charcoal** as base color
- Add subtle radial gradients for depth
- Background must be **LIGHTER than cards** (cards are darker and float on top)

```css
/* Source: stafull-design-system */
body {
  background:
    radial-gradient(ellipse at 30% 20%, #2a2b2f 0%, transparent 50%),
    radial-gradient(ellipse at 70% 80%, #252628 0%, transparent 50%),
    #1e1f22;  /* --sf-bg-page: LIGHTER base */
}
```

## 4.3 DO NOT
- ‚ùå Use blue, purple, slate, or cool gray tones
- ‚ùå Use pure flat black (#000000)
- ‚ùå Add texture, noise, or patterns (unless explicitly instructed)
- ‚ùå Let background compete with cards
- ‚ùå Use colored backgrounds
- ‚ùå Make background darker than cards (background is LIGHTER, cards are DARKER)

---

# SECTION 5: CARD MATERIAL RULES (CRITICAL)

## 5.1 Card Philosophy
**Cards are PHYSICAL OBJECTS, not containers.**

Every card must:
- Float above the background
- Have a visible bottom edge
- Feel heavier than the surface beneath it
- Fit their content ‚Äî expand when data increases
- Never clip content

## 5.2 Required Card Styling (Source: stafull-design-system)

### Border (Dark Theme)
```css
border: 1px solid rgba(255, 255, 255, 0.04);  /* Source: stafull-design-system */
```

### Border (Light Theme)
```css
border: 1px solid #d4d4d4;  /* --sf-gray-300 */
```

### Background (Satin Finish - DARKER than page)
```css
background: linear-gradient(
  180deg,
  var(--sf-card-top) 0%,    /* #131416 */
  var(--sf-card-bottom) 100%  /* #0a0b0d */
);
```

### Outer Shadow (3-layer heavy shadow)
```css
box-shadow:
  0 25px 50px rgba(0, 0, 0, 0.8),
  0 10px 20px rgba(0, 0, 0, 0.6),
  0 1px 0 rgba(255, 255, 255, 0.06) inset;
```

### Inner Highlight (Top sheen - 0.07 opacity)
```css
.sf-card::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 50%;
  background: linear-gradient(
    180deg,
    rgba(255, 255, 255, 0.07) 0%,
    transparent 100%
  );
  pointer-events: none;
  border-radius: var(--sf-radius-card) var(--sf-radius-card) 0 0;
}
```

### Border Radius
```css
border-radius: 14px;  /* --sf-radius-card */
```

## 5.3 Internal Dividers
```css
.sf-divider {
  height: 1px;
  margin: 12px 0;
  background: linear-gradient(
    to right,
    rgba(255, 255, 255, 0),
    rgba(255, 255, 255, 0.08),
    rgba(255, 255, 255, 0)
  );
}
```

## 5.4 DO
- Treat every card as a physical object
- Use subtle vertical gradients
- Use satin finish (not glossy on cards)
- Strong but controlled shadow separation
- Use 1px dividers (rgba 255,255,255,0.05‚Äì0.07) to separate content

## 5.5 DO NOT
- ‚ùå Use flat backgrounds for cards
- ‚ùå Let card edges disappear into background
- ‚ùå Make cards lighter than the background
- ‚ùå Reuse same gradient for background and cards
- ‚ùå Use flat rectangles with tiny shadow
- ‚ùå Let card bottoms visually merge together
- ‚ùå Let content touch card edges
- ‚ùå Float elements without anchors
- ‚ùå Stack unrelated data without separation

---

# SECTION 6: GLOSSY ELEMENTS (WHERE CLAUDE TYPICALLY FAILS)

## 6.1 What MUST Be Glossy
- Buttons
- Status pills / badges
- Progress bars
- Graph lines
- Accent highlights
- Indicators

## 6.2 Gloss Requirements
Every glossy element MUST have:
1. A **top highlight**
2. A **mid-tone body**
3. A **darker lower edge**
4. A **subtle shadow beneath**

**If it looks flat ‚Üí WRONG**

## 6.3 Gloss Utility Class (Source: stafull-design-system)
```css
.sf-glossy {
  position: relative;
  border-radius: inherit;
  box-shadow:
    0 8px 16px rgba(0, 0, 0, 0.5),
    0 1px 0 rgba(255, 255, 255, 0.2) inset;
  overflow: hidden;
}

.sf-glossy::before {
  content: "";
  position: absolute;
  inset: 0;
  background: linear-gradient(180deg, rgba(255, 255, 255, 0.3) 0%, rgba(255, 255, 255, 0.08) 40%, transparent 70%);
  pointer-events: none;
}
```

## 6.4 Color Fill Classes (Source: stafull-design-system)
```css
.sf-fill-red {
  background: linear-gradient(180deg, #c51616 0%, #750000 100%);
}
.sf-fill-green {
  background: linear-gradient(180deg, #45d975 0%, #1a8f40 100%);
}
.sf-fill-yellow {
  background: linear-gradient(180deg, #ffe066 0%, #d4a020 100%);
}
.sf-fill-blue {
  background: linear-gradient(180deg, #6aa3ff 0%, #2855c0 100%);
}
.sf-fill-grey {
  background: linear-gradient(180deg, #35363c 0%, #1a1b1f 100%);
}
```

## 6.5 DO
- Use color sparingly
- Use color to draw the eye
- Make all colored elements feel **coated or lacquered**
- Wrap glossy surfaces in shared classes
- Always compose with `sf-glossy` + `sf-fill-*`

## 6.6 DO NOT
- ‚ùå Apply gloss to cards or background (those are SATIN)
- ‚ùå Over-gloss (no glass, no neon)
- ‚ùå Fake gloss with blur or opacity alone
- ‚ùå Use flat color fills
- ‚ùå Use pastel or washed-out tones
- ‚ùå Let color dominate the UI
- ‚ùå Color large surfaces
- ‚ùå Use a single hex color for buttons, pills, or bars

---

# SECTION 7: COLOR SYSTEM

## 7.1 Brand Red (#a30000)
**Accent ONLY.** Used for:
- Primary actions
- Key indicators
- Active states
- Must always appear **GLOSSY**

## 7.2 Complete Color Tokens (Source: stafull-design-system)
```css
:root {
  /* Background (LIGHTER) + Cards (DARKER) */
  --sf-bg-page: #1e1f22;       /* Page background - LIGHTER */
  --sf-card-top: #131416;      /* Card gradient top - DARKER */
  --sf-card-bottom: #0a0b0d;   /* Card gradient bottom - DARKEST */

  /* Brand Red (accent only) */
  --sf-red: #a30000;
  --sf-red-light: #c51616;
  --sf-red-dark: #750000;

  /* Product Colors - Fuel Types (Source: stafull-design-system) */
  --sf-diesel: #31b34a;      /* Green */
  --sf-regular: #f2c94c;     /* Yellow */
  --sf-premium: #d06363;     /* Red/Mauve */
  --sf-def: #4a7ce8;         /* Blue */

  /* Status Colors (v1 Original) */
  --sf-success: #22c55e;
  --sf-warning: #eab308;
  --sf-error: #ef4444;
  --sf-info: #3b82f6;

  /* Card Border Colors (Source: stafull-design-system) */
  --sf-border-card-dark: rgba(255, 255, 255, 0.04);   /* Dark theme - subtle edge catch */
  --sf-border-card-light: #d4d4d4; /* Light theme card border */

  /* Neutral Grays (Satin Blacks) */
  --sf-gray-50: #fafafa;
  --sf-gray-100: #f5f5f5;
  --sf-gray-200: #e5e5e5;
  --sf-gray-300: #d4d4d4;
  --sf-gray-400: #a3a3a3;
  --sf-gray-500: #737373;
  --sf-gray-600: #525252;
  --sf-gray-700: #404040;
  --sf-gray-800: #262626;
  --sf-gray-850: #1f1f1f;
  --sf-gray-900: #171717;

  /* Typography */
  --sf-text-primary: #ffffff;
  --sf-text-secondary: rgba(255, 255, 255, 0.85);
  --sf-text-muted: rgba(255, 255, 255, 0.6);
}
```

## 7.3 Product Color Semantics (Source: stafull-design-system)
| Product | Color | Hex |
|---------|-------|-----|
| Diesel | Green | #31b34a |
| Regular | Yellow | #f2c94c |
| Premium | Red/Mauve | #d06363 |
| DEF | Blue | #4a7ce8 |

## 7.4 Status Color Semantics (v1 Original)
| Status | Color | Hex |
|--------|-------|-----|
| Success | Green | #22c55e |
| Warning | Yellow | #eab308 |
| Error | Red | #ef4444 |
| Info | Blue | #3b82f6 |

## 7.5 DO NOT
- ‚ùå Invent new semantic colors
- ‚ùå Reuse colors for different meanings
- ‚ùå Desaturate semantic colors
- ‚ùå Use blue, purple, slate as base UI colors

---

# SECTION 8: TYPOGRAPHY

## 8.1 Font Family
```css
font-family: 'Montserrat', system-ui, -apple-system, BlinkMacSystemFont, sans-serif;
```

## 8.2 Type Scale
```css
--sf-text-heading: 16px, weight 600;
--sf-text-subheading: 13px, weight 500;
--sf-text-body: 13px, weight 400;
--sf-text-muted: 12px, weight 400;
```

## 8.3 Typography Rules
| Element | Size | Weight | Color |
|---------|------|--------|-------|
| Heading | 16px | 600 (semibold) | --sf-text-primary |
| Subheading | 13px | 500 (medium) | --sf-text-secondary |
| Body | 13px | 400 (regular) | --sf-text-secondary |
| Muted | 12px | 400 (regular) | --sf-text-muted |
| Numeric Values | Larger | Heavier | High contrast |

## 8.4 DO
- Primary text: **pure white**
- Secondary text: **near-white** (still high contrast)
- Numeric values: Larger, Heavier, Confident
- Clear hierarchy
- High contrast against background

## 8.5 DO NOT
- ‚ùå **ALL CAPS** (for anything, ever)
- ‚ùå Dim text for "style"
- ‚ùå Light gray on dark backgrounds
- ‚ùå Shrink text to fit bad layouts
- ‚ùå Mix font styles
- ‚ùå Use decorative fonts
- ‚ùå Center body text arbitrarily
- ‚ùå Shrink text below 12px for body content

---

# SECTION 9: BUTTONS

## 9.1 Button Base (Source: stafull-design-system)
```css
.sf-btn {
  border-radius: 8px;  /* --sf-radius-button */
  padding: 10px 18px;
  border: none;
  font-size: 13px;
  font-weight: 600;
  color: var(--sf-text-primary);
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  gap: 6px;
}
```

## 9.2 Button Variants
```html
<!-- Primary (Red, Glossy) -->
<button class="sf-btn sf-glossy sf-fill-red">confirm delivery</button>

<!-- Secondary (Gray, Glossy) -->
<button class="sf-btn sf-glossy sf-fill-grey">save draft</button>

<!-- Outline -->
<button class="sf-btn-outline">view all</button>
```

## 9.3 Outline Button (Source: stafull-design-system)
```css
.sf-btn-outline {
  border-radius: 8px;
  padding: 10px 18px;
  background: transparent;
  border: 1px solid rgba(255, 255, 255, 0.2);
  color: var(--sf-text-secondary);
}
```

## 9.4 DO NOT
- ‚ùå ALL CAPS button text
- ‚ùå Custom button shadows/backgrounds per button
- ‚ùå Flat colored buttons without sf-glossy

---

# SECTION 10: STATUS PILLS & BADGES

## 10.1 Pill Base (Source: stafull-design-system)
```css
.sf-pill {
  border-radius: 999px;  /* --sf-radius-pill */
  font-size: 10px;
  padding: 4px 10px;
  font-weight: 600;
  text-transform: none;
  display: inline-flex;
  align-items: center;
  gap: 4px;
  color: var(--sf-text-primary);
}
```

## 10.2 Usage
```html
<span class="sf-pill sf-glossy sf-fill-green">on schedule</span>
<span class="sf-pill sf-glossy sf-fill-yellow">delayed</span>
<span class="sf-pill sf-glossy sf-fill-red">urgent</span>
<span class="sf-pill sf-glossy sf-fill-blue">info</span>
```

## 10.3 DO NOT
- ‚ùå Emojis in pills
- ‚ùå Flat pills (no gloss)
- ‚ùå Dark text on dark pills

---

# SECTION 11: PROGRESS BARS

## 11.1 Structure (Source: stafull-design-system)
```css
.sf-progress-track {
  height: 10px;
  border-radius: 999px;
  background: radial-gradient(ellipse at top, #252628 0%, #0d0e10 100%);
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.8) inset;
  overflow: hidden;
}

.sf-progress-fill {
  height: 100%;
  border-radius: inherit;
  /* Apply sf-glossy + sf-fill-* */
}
```

## 11.2 Usage
```html
<div class="sf-progress-track">
  <div class="sf-progress-fill sf-glossy sf-fill-green" style="width: 68%;"></div>
</div>
```

---

# SECTION 12: MAP CARD RULES (ABSOLUTE)

## 12.1 Map Requirements
- Maps MUST be **landscape orientation**
- Minimum aspect ratio: **3:2**
- Preferred aspect ratio: **16:9**
- Map must feel like a **"command surface"**
- Map is a **dominant element** ‚Äî justify size with information

## 12.2 Map Card CSS
```css
.sf-card-map {
  display: flex;
  flex-direction: column;
  gap: 12px;
  min-height: 320px;
  aspect-ratio: 16 / 9;
}

.sf-map-surface {
  flex: 1;
  border-radius: calc(var(--sf-radius-card) - 8px);
  background:
    linear-gradient(to right, rgba(0,0,0,0.08) 1px, transparent 1px),
    linear-gradient(to bottom, rgba(0,0,0,0.08) 1px, transparent 1px);
  background-size: 28px 28px;
  background-color: #dcdfe4;
  position: relative;
  overflow: hidden;
}
```

## 12.3 DO NOT
- ‚ùå Portrait maps
- ‚ùå Shrink map to fit grid
- ‚ùå Treat map as secondary content
- ‚ùå Thumbnail-sized maps

---

# SECTION 13: ICONS & SYMBOLS

## 13.1 DO
- Use custom, minimal glyphs
- Use monochrome icons
- Keep icons secondary to text
- Simple neutral shapes (circles, chevrons)
- SVG-style, monochrome
- Tie icons to status/product colors or neutrals

## 13.2 DO NOT
- ‚ùå **EMOJIS** (üöö, ‚ö†Ô∏è, ‚úÖ, etc.) ‚Äî NEVER
- ‚ùå Default icon packs
- ‚ùå Font Awesome fallbacks
- ‚ùå Mix icon styles
- ‚ùå Color icons randomly

---

# SECTION 14: SPACING SYSTEM

## 14.1 Donald's Original Spacing Scale (v1)
```css
/* 4 / 6 / 10 / 14 / 18 / 30 / 45 / 90 */
--sf-spacing-xs: 4px;
--sf-spacing-sm: 6px;
--sf-spacing-md: 10px;
--sf-spacing-lg: 14px;
--sf-spacing-xl: 18px;
--sf-spacing-2xl: 30px;
--sf-spacing-3xl: 45px;
--sf-spacing-4xl: 90px;
```

## 14.2 DO
- Use consistent spacing increments
- Cards should "nest" together visually
- Vertical rhythm matters more than grid perfection
- Interior spacing should breathe
- Exterior spacing should be tight but clear
- Cards should feel: Dense, Intentional, Premium

## 14.3 DO NOT
- ‚ùå Leave large empty gutters
- ‚ùå Space cards evenly "just because"
- ‚ùå Random padding values
- ‚ùå Large empty gaps
- ‚ùå Crowd text
- ‚ùå Rely on padding to fix bad sizing

---

# SECTION 15: BORDER RADIUS

## 15.1 Radius Scale (Source: stafull-design-system)
```css
--sf-radius-sm: 4px;
--sf-radius-md: 8px;
--sf-radius-button: 8px;  /* squared, not pill */
--sf-radius-card: 14px;   /* Source: stafull-design-system */
--sf-radius-pill: 999px;
--sf-radius-lg: 20px;
```

## 15.2 Usage
| Element | Radius |
|---------|--------|
| Cards | 14px |
| Buttons | 8px |
| Pills/Badges | 999px (full) |
| Inputs | 8px |
| Small elements | 4px |

---

# SECTION 16: SHADOWS

## 16.1 Shadow Scale (Source: stafull-design-system)
```css
/* Strong shadow for cards - 3-layer heavy shadow */
--sf-shadow-card:
  0 25px 50px rgba(0, 0, 0, 0.8),
  0 10px 20px rgba(0, 0, 0, 0.6),
  0 1px 0 rgba(255, 255, 255, 0.06) inset;

/* Soft shadow */
--sf-shadow-soft:
  0 10px 25px rgba(0, 0, 0, 0.65),
  0 1px 0 rgba(255, 255, 255, 0.04) inset;

/* Glossy element shadow (Source: stafull-design-system) */
--sf-shadow-glossy:
  0 8px 16px rgba(0, 0, 0, 0.5),
  0 1px 0 rgba(255, 255, 255, 0.2) inset;

/* Button shadow (Source: stafull-design-system) */
--sf-shadow-button:
  0 8px 16px rgba(0, 0, 0, 0.5),
  0 1px 0 rgba(255, 255, 255, 0.2) inset;
```

## 16.2 Rules
- Cards must visually lift off background
- Bottom edges must be readable
- Use shadow + gradient contrast together
- Shadows are **BLACK ONLY** ‚Äî no colored shadows

## 16.3 DO NOT
- ‚ùå Rely on shadow alone
- ‚ùå Allow card bottoms to merge visually
- ‚ùå Flatten depth for "cleanliness"
- ‚ùå Use red/colored shadows

---

# SECTION 17: TRANSITIONS & ANIMATION

## 17.1 Animation Timing
```css
--sf-transition-fast: 100ms ease;
--sf-transition-base: 200ms ease;
--sf-transition-slow: 300ms ease;
```

## 17.2 Allowed Durations
- 100ms
- 200ms
- 300ms

## 17.3 DO NOT
- ‚ùå 150ms, 250ms, 350ms, 400ms, 500ms+

---

# SECTION 18: FOCUS & HOVER STATES

## 18.1 Focus Ring (MANDATORY)
```css
--sf-focus-ring: 1px solid var(--sf-red);
--sf-focus-offset: 0px;
```

## 18.2 Focus-Visible Usage
```css
:focus-visible {
  outline: 1px solid var(--sf-red);
  outline-offset: 0;
}
```

## 18.3 Hover States (MANDATORY)
All interactive elements MUST have a hover outline:
```css
.element:hover {
  outline: 1px solid var(--sf-red);
  outline-offset: 0;
}
```

## 18.4 Rules
| Property | Value | Reason |
|----------|-------|--------|
| outline width | **1px** | Never 2px or more |
| outline-offset | **0** | Never 2px or more |
| outline color | `var(--sf-red)` or `#a30000` | Brand red accent |

## 18.5 DO NOT
- ‚ùå `outline: 2px` (always 1px)
- ‚ùå `outline-offset: 2px` (always 0)
- ‚ùå Focus rings with glow/box-shadow instead of outline
- ‚ùå Colored box-shadows for focus (use outline only)

---

# SECTION 19: DATA VISUALIZATION

## 19.1 Graphs
- Clean
- Correct scale
- Purposeful
- **Gloss-accented lines**

## 19.2 DO NOT
- ‚ùå Placeholder-looking graphs
- ‚ùå Distort aspect ratios
- ‚ùå Overdecorate charts

---

# SECTION 20: CARD CONTENT & SIZING (CONTENT-DRIVEN)

## 20.1 ABSOLUTE RULE
**Cards resize to content ‚Äî NEVER the opposite.**

## 20.2 DO
- Grow cards when content grows
- Shrink cards when content is minimal
- Resize maps to be dominant elements
- Resize graphs to show meaningful data
- Every card should answer a question

## 20.3 DO NOT
- ‚ùå Stretch cards vertically to fill space
- ‚ùå Leave cards half-empty
- ‚ùå Cram data to maintain symmetry
- ‚ùå Show low-importance data just to fill space
- ‚ùå Leave placeholder text
- ‚ùå Leave cards with one lonely number
- ‚ùå Create demo cards with empty space

---

# SECTION 21: MOCK DATA REQUIREMENTS

## 21.1 Every Card Must Contain
- Labels
- Values
- Secondary details

## 21.2 DO
- Use believable numbers and names
- Repeat patterns across cards
- Fill every card with data

## 21.3 DO NOT
- ‚ùå Lorem ipsum
- ‚ùå Empty lists
- ‚ùå "Example" labels

---

# SECTION 22: DASHBOARD CONTENT REQUIREMENTS (MINIMUM)

## 22.1 Left Zone (Design System)
- Color palette card
- Typography card
- Buttons card
- Status indicators card

## 22.2 Center Zone (Core Dashboard)
- Wide map card (landscape)
- Row of 3 KPI cards
- Transactions + mini trend card
- Top drivers / leaderboard card

## 22.3 Right Zone (Status & Resources)
- Fuel inventory card (progress bars)
- Recent alerts card

**Total: At least 12 cards minimum**

---

# SECTION 23: THEME VARIANTS

**ALL pages support BOTH dark and light themes.** This is not limited to internal vs external pages.

## 23.1 Dark Theme (Primary)
- Background: Satin black gradients
- Cards: Darker than background
- Text: White/near-white

## 23.2 Light Theme
- Background: Light gray/white satin
- Cards: White with subtle shadows
- Text: Dark gray/black
- Same glossy rules apply to buttons/pills

## 23.3 Theme Implementation
- Every portal page MUST support theme toggling
- Theme preference should persist per user
- Components should use CSS custom properties for theme-aware colors
- Test both themes before shipping

---

# SECTION 24: FINAL VALIDATION CHECKLIST

Before ANYTHING is considered "done":

- [ ] Cards feel tangible (not flat rectangles)
- [ ] Gloss exists on buttons, pills, progress bars
- [ ] Text is bright and readable
- [ ] Layout feels intentional (not template-like)
- [ ] No card looks empty
- [ ] No card looks cramped
- [ ] Map feels important and is landscape
- [ ] UI does NOT look like a template
- [ ] No emojis anywhere
- [ ] No ALL CAPS anywhere
- [ ] Brand spelled correctly (StƒÅFull)
- [ ] 3-column layout respected
- [ ] No scrolling required
- [ ] All colored elements use sf-glossy + sf-fill-*

**If ANY item fails ‚Üí REVISE before shipping.**

---

# SECTION 25: FORBIDDEN LIST (QUICK REFERENCE)

## Typography
- ‚ùå ALL CAPS (`text-transform: uppercase` - use `none`)
- ‚ùå Font weight < 400 or > 700
- ‚ùå Decorative fonts
- ‚ùå Text smaller than 12px body

## Colors
- ‚ùå Blue/purple/slate as base UI
- ‚ùå Pure #000000 backgrounds
- ‚ùå Colored shadows (red rgba in shadows)
- ‚ùå Pastel/washed-out tones
- ‚ùå Flat color fills on interactive elements

## Layout
- ‚ùå Single column
- ‚ùå Portrait orientation
- ‚ùå Scrolling dashboards
- ‚ùå Empty space
- ‚ùå Almost-aligned cards

## Elements
- ‚ùå Emojis
- ‚ùå Default icon packs
- ‚ùå Flat buttons/pills
- ‚ùå Portrait maps

## Focus & Hover
- ‚ùå `outline: 2px` (must be 1px)
- ‚ùå `outline-offset: 2px` (must be 0)
- ‚ùå Colored focus shadows (use outline only)
- ‚ùå Red rgba in shadows (shadows must be black only)

## Spacing
- ‚ùå Random padding values
- ‚ùå Large empty gutters

## Brand
- ‚ùå STAFULL, Stafull, staFull
- ‚ùå Any spelling without macron (unless technically required)

---

**END OF DESIGN RULES v2.0**
