// @stafull/shared/src/components/atoms/StaFullLogo/StaFullLogo.jsx
/**
 * StaFullLogo Atom
 * Brand logo with macron styling
 * CSS-based wordmark with configurable size
 */

import styles from './StaFullLogo.module.css';
import { cn } from '../../../utils/cn';

const SIZES = {
  xs: 18,
  sm: 24,
  md: 32,
  lg: 40,
  xl: 48,
  '2xl': 64,
};

export function StaFullLogo({
  size = 'md',
  color,
  className,
  onClick,
  linkToHome = false,
  ...props
}) {
  const fontSize = typeof size === 'number' ? size : SIZES[size] || SIZES.md;

  const handleClick = (e) => {
    if (onClick) {
      onClick(e);
    } else if (linkToHome && typeof window !== 'undefined') {
      window.location.href = '/';
    }
  };

  const logoStyle = {
    fontSize: `${fontSize}px`,
    ...(color && { color }),
  };

  return (
    <span
      className={cn(
        styles.logo,
        linkToHome && styles.clickable,
        className
      )}
      style={logoStyle}
      onClick={linkToHome || onClick ? handleClick : undefined}
      role={linkToHome ? 'button' : undefined}
      tabIndex={linkToHome ? 0 : undefined}
      onKeyDown={linkToHome ? (e) => e.key === 'Enter' && handleClick(e) : undefined}
      {...props}
    >
      St<span className={styles.macronA}>ā</span><span className={styles.full}>Full</span>
    </span>
  );
}

export default StaFullLogo;
