// @stafull/shared/src/tokens/index.js
/**
 * StaFull Design Tokens
 * FINALIZED by Donald B. Havery - DO NOT MODIFY without approval
 *
 * These tokens are the single source of truth for the entire design system.
 */

// =============================================================================
// COLORS
// =============================================================================

export const colors = {
  // Primary Red Palette (GLOSSY appearance - NO glow effects)
  red: {
    primary: '#a30000',    // Main brand color
    bright: '#cc0000',     // Hover states, emphasis
    dark: '#8a0000',       // Pressed states, depth
  },

  // Standard Red Gradient
  redGradient: 'linear-gradient(160deg, #cc0000, #a30000, #8a0000)',

  // Fuel Type Colors
  fuel: {
    diesel: '#487800',     // Green
    regular: '#fff34a',    // Yellow
    premium: '#965154',    // Mauve
    def: '#4371ba',        // Blue
  },

  // Neutral Colors
  neutral: {
    background: '#0a0a0a', // Main dark background
    surface: '#1a1a1a',    // Cards, panels
    surfaceHover: '#252525', // Card hover state
    border: '#2a2a2a',     // Dividers, borders
    borderLight: '#3a3a3a', // Lighter borders
  },

  // Text Colors
  text: {
    primary: '#ffffff',    // Main text
    secondary: '#a0a0a0',  // Muted text
    tertiary: '#707070',   // Disabled/placeholder
    inverse: '#0a0a0a',    // Text on light backgrounds
  },

  // Status Colors
  status: {
    success: '#22c55e',
    warning: '#f59e0b',
    error: '#ef4444',
    info: '#3b82f6',
  },
};

// =============================================================================
// SPACING (Donald's Original Scale)
// =============================================================================

export const spacing = {
  1: '4px',   // xxs - tight spacing, fine adjustments
  2: '6px',   // xs - small gaps between elements
  3: '10px',  // sm - icon gaps, small padding
  4: '14px',  // md - standard padding, margins
  5: '18px',  // lg - card padding, section gaps
  6: '30px',  // xl - medium section spacing
  7: '45px',  // xxl - large section spacing
  8: '90px',  // xxxl - page section breaks
};

// Semantic spacing aliases
export const spacingAliases = {
  none: '0',
  xxs: spacing[1],
  xs: spacing[2],
  sm: spacing[3],
  md: spacing[4],
  lg: spacing[5],
  xl: spacing[6],
  xxl: spacing[7],
  xxxl: spacing[8],
};

// =============================================================================
// TYPOGRAPHY
// =============================================================================

export const typography = {
  // Font Family - ONLY Montserrat
  fontFamily: "'Montserrat', sans-serif",

  // Font Weights
  weights: {
    regular: 400,  // Body text
    medium: 500,   // Labels, navigation
    semibold: 600, // Subheadings
    bold: 700,     // Headings, emphasis
  },

  // Font Sizes (rem based)
  sizes: {
    xs: '0.75rem',     // 12px - Small labels, captions
    sm: '0.875rem',    // 14px - Secondary text
    base: '1rem',      // 16px - Body text
    lg: '1.125rem',    // 18px - Large body
    xl: '1.25rem',     // 20px - Subheadings
    '2xl': '1.5rem',   // 24px - Headings
    '3xl': '2rem',     // 32px - Page titles
    '4xl': '2.5rem',   // 40px - Hero text
  },

  // Line Heights
  lineHeights: {
    tight: 1.2,
    normal: 1.5,
    relaxed: 1.75,
  },
};

// =============================================================================
// BORDER RADIUS (Donald's Original)
// =============================================================================

export const borderRadius = {
  none: '0',
  sm: '4px',     // Inputs, small buttons
  md: '8px',     // Cards, modals
  lg: '12px',    // Large containers
  xl: '16px',    // Extra large containers
  full: '50px',  // Pills, rounded elements
  circle: '50%', // True circular elements
};

// =============================================================================
// ANIMATION TIMING
// =============================================================================

export const animation = {
  // Allowed durations ONLY - NO other values
  fast: '100ms',    // Quick micro-interactions
  normal: '200ms',  // Standard transitions
  slow: '300ms',    // Larger animations

  // Easing functions
  easing: {
    default: 'ease',
    easeIn: 'ease-in',
    easeOut: 'ease-out',
    easeInOut: 'ease-in-out',
  },
};

// =============================================================================
// FOCUS RINGS
// =============================================================================

export const focus = {
  // Focus ring - EXACTLY 1px (NOT 2px or 3px)
  width: '1px',
  style: 'solid',
  color: colors.red.primary,
  offset: '2px',
};

// =============================================================================
// SHADOWS
// =============================================================================

export const shadows = {
  none: 'none',
  subtle: '0 2px 4px rgba(0, 0, 0, 0.1)',
  medium: '0 4px 8px rgba(0, 0, 0, 0.15)',
  prominent: '0 8px 16px rgba(0, 0, 0, 0.2)',
  elevated: '0 12px 24px rgba(0, 0, 0, 0.25)',
};

// =============================================================================
// BREAKPOINTS
// =============================================================================

export const breakpoints = {
  sm: '640px',
  md: '768px',
  lg: '1024px',
  xl: '1280px',
  '2xl': '1440px',
  '3xl': '1920px',
};

// =============================================================================
// Z-INDEX SCALE
// =============================================================================

export const zIndex = {
  base: 0,
  dropdown: 100,
  sticky: 200,
  fixed: 300,
  modalBackdrop: 400,
  modal: 500,
  popover: 600,
  tooltip: 700,
  toast: 800,
};

// =============================================================================
// LAYOUT
// =============================================================================

export const layout = {
  sidebar: {
    width: '280px',
    collapsedWidth: '72px',
  },
  header: {
    height: '64px',
  },
  maxWidth: {
    content: '1400px',
    form: '600px',
  },
};

// =============================================================================
// DEFAULT EXPORT
// =============================================================================

const tokens = {
  colors,
  spacing,
  spacingAliases,
  typography,
  borderRadius,
  animation,
  focus,
  shadows,
  breakpoints,
  zIndex,
  layout,
};

export default tokens;
