// @stafull/shared/src/contexts/AuthContext.jsx
/**
 * AuthContext - Unified authentication state management
 *
 * Provides:
 * - User authentication state
 * - Token management
 * - Session timeout for internal users
 * - Activity tracking
 */

import { createContext, useContext, useState, useEffect, useCallback, useRef } from 'react';
import authService from '../services/auth';

const AuthContext = createContext(null);

// Internal roles that require session timeout
const INTERNAL_ROLES = ['holdings_admin', 'franchise_owner', 'manager', 'driver'];
const SESSION_TIMEOUT_MS = 30 * 60 * 1000; // 30 minutes
const WARNING_TIME_MS = 25 * 60 * 1000; // 25 minutes
const HEARTBEAT_INTERVAL_MS = 60 * 1000; // 1 minute

// Owner email - indefinite session until launch
const OWNER_EMAIL = 'dbhavery@stafull.com';

// Secondary auth timeout (HR, Finance, Dev, Site Editor)
const SECONDARY_TIMEOUT_MS = 15 * 60 * 1000; // 15 minutes
const SECONDARY_WARNING_MS = 12 * 60 * 1000; // 12 minutes

// Dev mode bypass - check URL param and environment
const isDevBypass = () => {
  if (typeof window === 'undefined') return false;
  const params = new URLSearchParams(window.location.search);
  return import.meta.env.DEV && params.get('devBypass') === 'true';
};

// Mock user for dev bypass based on URL param
const getDevBypassUser = () => {
  if (typeof window === 'undefined') return null;
  const params = new URLSearchParams(window.location.search);
  const role = params.get('role') || 'holdings_admin';

  return {
    id: 'dev-user-001',
    email: 'dev@stafull.com',
    firstName: 'Dev',
    lastName: 'User',
    role,
    termsAccepted: true,
    emailVerified: true,
  };
};

export function AuthProvider({ children, onSessionExpired, onSessionWarning }) {
  const [user, setUser] = useState(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);
  const [sessionWarning, setSessionWarning] = useState(false);

  // Secondary auth state (for HR, Finance, Dev, Site Editor)
  const [secondaryAuth, setSecondaryAuth] = useState(null);
  const [secondaryWarning, setSecondaryWarning] = useState(false);

  const lastActivityRef = useRef(Date.now());
  const timeoutRef = useRef(null);
  const warningRef = useRef(null);
  const heartbeatRef = useRef(null);
  const secondaryTimeoutRef = useRef(null);
  const secondaryWarningRef = useRef(null);
  const secondaryLastActivityRef = useRef(Date.now());

  // Check if user is the owner (indefinite session until launch)
  const isOwner = user && user.email === OWNER_EMAIL;

  // Check if user requires session timeout (not owner, and is internal role)
  const requiresTimeout = user && INTERNAL_ROLES.includes(user.role) && !isOwner;

  // Reset activity timer
  const resetActivityTimer = useCallback(() => {
    lastActivityRef.current = Date.now();
    setSessionWarning(false);

    // Clear existing timers
    if (timeoutRef.current) clearTimeout(timeoutRef.current);
    if (warningRef.current) clearTimeout(warningRef.current);

    if (!requiresTimeout) return;

    // Set warning timer
    warningRef.current = setTimeout(() => {
      setSessionWarning(true);
      onSessionWarning?.();
    }, WARNING_TIME_MS);

    // Set timeout timer
    timeoutRef.current = setTimeout(() => {
      handleSessionExpired();
    }, SESSION_TIMEOUT_MS);
  }, [requiresTimeout, onSessionWarning]);

  // Handle session expiry
  const handleSessionExpired = useCallback(async () => {
    setSessionWarning(false);
    await authService.logout();
    setUser(null);
    onSessionExpired?.();
  }, [onSessionExpired]);

  // Track user activity
  const trackActivity = useCallback(() => {
    if (requiresTimeout) {
      resetActivityTimer();
    }
  }, [requiresTimeout, resetActivityTimer]);

  // Initialize auth state
  const initAuth = useCallback(async () => {
    setLoading(true);
    setError(null);

    try {
      // Check for dev bypass mode
      if (isDevBypass()) {
        const devUser = getDevBypassUser();
        console.log('[AUTH] Dev bypass mode - using mock user:', devUser.role);
        setUser(devUser);
        setLoading(false);
        return;
      }

      const tokens = authService.getStoredTokens();

      if (!tokens.accessToken) {
        setUser(null);
        setLoading(false);
        return;
      }

      // Get session status
      const { ok, data } = await authService.getSessionStatus();

      if (ok && data.authenticated && data.valid) {
        setUser(data.user);

        // Start session timeout if needed
        if (INTERNAL_ROLES.includes(data.user.role)) {
          resetActivityTimer();
        }
      } else {
        // Session invalid - clear tokens
        authService.clearTokens();
        setUser(null);

        if (data.sessionExpired) {
          onSessionExpired?.();
        }
      }
    } catch (err) {
      console.error('[AUTH] Init error:', err);
      setError('Failed to initialize authentication');
      setUser(null);
    } finally {
      setLoading(false);
    }
  }, [resetActivityTimer, onSessionExpired]);

  // Login
  const login = useCallback(async (email, password) => {
    setLoading(true);
    setError(null);

    try {
      const { ok, data } = await authService.login(email, password);

      if (ok && data.success) {
        setUser(data.user);

        // Start session timeout if needed
        if (INTERNAL_ROLES.includes(data.user.role)) {
          resetActivityTimer();
        }

        return { success: true, data };
      } else {
        setError(data.error || 'Login failed');
        return { success: false, error: data.error, code: data.code };
      }
    } catch (err) {
      console.error('[AUTH] Login error:', err);
      setError('Network error. Please try again.');
      return { success: false, error: 'Network error' };
    } finally {
      setLoading(false);
    }
  }, [resetActivityTimer]);

  // Register
  const register = useCallback(async (userData) => {
    setLoading(true);
    setError(null);

    try {
      const { ok, data } = await authService.register(userData);

      if (ok && data.success) {
        return { success: true, data };
      } else {
        setError(data.error || 'Registration failed');
        return { success: false, error: data.error, code: data.code };
      }
    } catch (err) {
      console.error('[AUTH] Register error:', err);
      setError('Network error. Please try again.');
      return { success: false, error: 'Network error' };
    } finally {
      setLoading(false);
    }
  }, []);

  // Logout
  const logout = useCallback(async () => {
    // Clear timers
    if (timeoutRef.current) clearTimeout(timeoutRef.current);
    if (warningRef.current) clearTimeout(warningRef.current);
    if (heartbeatRef.current) clearInterval(heartbeatRef.current);

    await authService.logout();
    setUser(null);
    setSessionWarning(false);
  }, []);

  // Verify email
  const verifyEmail = useCallback(async (email, code) => {
    try {
      const { ok, data } = await authService.verifyEmail(email, code);
      return { success: ok && data.success, data };
    } catch (err) {
      console.error('[AUTH] Verify error:', err);
      return { success: false, error: 'Verification failed' };
    }
  }, []);

  // Accept terms
  const acceptTerms = useCallback(async () => {
    try {
      const { ok, data } = await authService.acceptTerms();

      if (ok && data.success) {
        setUser((prev) => prev ? { ...prev, termsAccepted: true } : null);
      }

      return { success: ok && data.success, data };
    } catch (err) {
      console.error('[AUTH] Accept terms error:', err);
      return { success: false, error: 'Failed to accept terms' };
    }
  }, []);

  // Extend session (reset timeout)
  const extendSession = useCallback(async () => {
    if (!requiresTimeout) return;

    try {
      await authService.heartbeat();
      resetActivityTimer();
    } catch (err) {
      console.error('[AUTH] Heartbeat error:', err);
    }
  }, [requiresTimeout, resetActivityTimer]);

  // Secondary auth - authenticate for sensitive areas (HR, Finance, Dev, Site Editor)
  const secondaryLogin = useCallback(async (password, area) => {
    // Verify password against current user's credentials
    if (!user) return { success: false, error: 'Not authenticated' };

    try {
      const { ok, data } = await authService.verifyPassword(password);

      if (ok && data.success) {
        const authData = { area, timestamp: Date.now() };
        setSecondaryAuth(authData);
        secondaryLastActivityRef.current = Date.now();

        // Set secondary timeout (15 min) - owner is exempt
        if (!isOwner) {
          if (secondaryTimeoutRef.current) clearTimeout(secondaryTimeoutRef.current);
          if (secondaryWarningRef.current) clearTimeout(secondaryWarningRef.current);

          secondaryWarningRef.current = setTimeout(() => {
            setSecondaryWarning(true);
          }, SECONDARY_WARNING_MS);

          secondaryTimeoutRef.current = setTimeout(() => {
            setSecondaryAuth(null);
            setSecondaryWarning(false);
          }, SECONDARY_TIMEOUT_MS);
        }

        return { success: true, area };
      } else {
        return { success: false, error: data.error || 'Invalid password' };
      }
    } catch (err) {
      console.error('[AUTH] Secondary auth error:', err);
      return { success: false, error: 'Verification failed' };
    }
  }, [user, isOwner]);

  // Reset secondary auth timer on activity
  const resetSecondaryTimer = useCallback(() => {
    if (!secondaryAuth || isOwner) return;

    secondaryLastActivityRef.current = Date.now();
    setSecondaryWarning(false);

    if (secondaryTimeoutRef.current) clearTimeout(secondaryTimeoutRef.current);
    if (secondaryWarningRef.current) clearTimeout(secondaryWarningRef.current);

    secondaryWarningRef.current = setTimeout(() => {
      setSecondaryWarning(true);
    }, SECONDARY_WARNING_MS);

    secondaryTimeoutRef.current = setTimeout(() => {
      setSecondaryAuth(null);
      setSecondaryWarning(false);
    }, SECONDARY_TIMEOUT_MS);
  }, [secondaryAuth, isOwner]);

  // Logout from secondary auth
  const secondaryLogout = useCallback(() => {
    if (secondaryTimeoutRef.current) clearTimeout(secondaryTimeoutRef.current);
    if (secondaryWarningRef.current) clearTimeout(secondaryWarningRef.current);
    setSecondaryAuth(null);
    setSecondaryWarning(false);
  }, []);

  // Check if secondary auth is valid for a specific area
  const hasSecondaryAccess = useCallback((area) => {
    if (isOwner) return true; // Owner has unlimited access
    return secondaryAuth && secondaryAuth.area === area;
  }, [secondaryAuth, isOwner]);

  // Setup activity listeners
  useEffect(() => {
    if (!requiresTimeout) return;

    const events = ['mousedown', 'keydown', 'touchstart', 'scroll'];

    const handleActivity = () => {
      trackActivity();
    };

    events.forEach((event) => {
      window.addEventListener(event, handleActivity, { passive: true });
    });

    return () => {
      events.forEach((event) => {
        window.removeEventListener(event, handleActivity);
      });
    };
  }, [requiresTimeout, trackActivity]);

  // Setup heartbeat for internal users
  useEffect(() => {
    if (!user || !requiresTimeout) return;

    heartbeatRef.current = setInterval(async () => {
      try {
        await authService.heartbeat();
      } catch (err) {
        console.error('[AUTH] Heartbeat failed:', err);
      }
    }, HEARTBEAT_INTERVAL_MS);

    return () => {
      if (heartbeatRef.current) {
        clearInterval(heartbeatRef.current);
      }
    };
  }, [user, requiresTimeout]);

  // Initialize on mount
  useEffect(() => {
    initAuth();

    return () => {
      if (timeoutRef.current) clearTimeout(timeoutRef.current);
      if (warningRef.current) clearTimeout(warningRef.current);
      if (heartbeatRef.current) clearInterval(heartbeatRef.current);
    };
  }, []);

  const value = {
    // State
    user,
    loading,
    error,
    isAuthenticated: !!user,
    sessionWarning,
    requiresTimeout,
    isOwner,

    // Secondary auth state
    secondaryAuth,
    secondaryWarning,

    // Actions
    login,
    logout,
    register,
    verifyEmail,
    acceptTerms,
    extendSession,
    refreshAuth: initAuth,

    // Secondary auth actions
    secondaryLogin,
    secondaryLogout,
    hasSecondaryAccess,
    resetSecondaryTimer,

    // Utilities
    getPortalUrl: authService.getPortalUrl,
  };

  return (
    <AuthContext.Provider value={value}>
      {children}
    </AuthContext.Provider>
  );
}

export function useAuth() {
  const context = useContext(AuthContext);

  if (!context) {
    throw new Error('useAuth must be used within an AuthProvider');
  }

  return context;
}

export default AuthContext;
