// @stafull/shared/src/services/auth.js
/**
 * Auth Service - API calls for authentication
 */

const API_BASE = import.meta.env.VITE_API_URL || 'https://api.stafull.com';
const TOKEN_KEY = 'stafull_token';
const REFRESH_KEY = 'stafull_refresh';

/**
 * Get stored tokens
 */
export function getStoredTokens() {
  return {
    accessToken: localStorage.getItem(TOKEN_KEY),
    refreshToken: localStorage.getItem(REFRESH_KEY),
  };
}

/**
 * Store tokens
 */
export function storeTokens(accessToken, refreshToken) {
  if (accessToken) localStorage.setItem(TOKEN_KEY, accessToken);
  if (refreshToken) localStorage.setItem(REFRESH_KEY, refreshToken);
}

/**
 * Clear tokens
 */
export function clearTokens() {
  localStorage.removeItem(TOKEN_KEY);
  localStorage.removeItem(REFRESH_KEY);
}

/**
 * Make authenticated API request
 */
async function authFetch(endpoint, options = {}) {
  const { accessToken } = getStoredTokens();

  const headers = {
    'Content-Type': 'application/json',
    ...options.headers,
  };

  if (accessToken) {
    headers.Authorization = `Bearer ${accessToken}`;
  }

  const response = await fetch(`${API_BASE}${endpoint}`, {
    ...options,
    headers,
  });

  // Handle token expiry
  if (response.status === 401) {
    const data = await response.json();
    if (data.code === 'INVALID_TOKEN' || data.code === 'SESSION_TIMEOUT') {
      // Try to refresh token
      const refreshed = await refreshAccessToken();
      if (refreshed) {
        // Retry with new token
        const newTokens = getStoredTokens();
        headers.Authorization = `Bearer ${newTokens.accessToken}`;
        return fetch(`${API_BASE}${endpoint}`, { ...options, headers });
      }
    }
  }

  return response;
}

/**
 * Login with email and password
 */
export async function login(email, password) {
  const response = await fetch(`${API_BASE}/api/auth/login`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ email, password }),
  });

  const data = await response.json();

  if (response.ok && data.tokens) {
    storeTokens(data.tokens.accessToken, data.tokens.refreshToken);
  }

  return { ok: response.ok, data };
}

/**
 * Register new user
 */
export async function register(userData) {
  const response = await fetch(`${API_BASE}/api/auth/register`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(userData),
  });

  return { ok: response.ok, data: await response.json() };
}

/**
 * Send verification code
 */
export async function sendVerification(email) {
  const response = await fetch(`${API_BASE}/api/auth/send-verification`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ email }),
  });

  return { ok: response.ok, data: await response.json() };
}

/**
 * Verify email with code
 */
export async function verifyEmail(email, code) {
  const response = await fetch(`${API_BASE}/api/auth/verify-email`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ email, code }),
  });

  return { ok: response.ok, data: await response.json() };
}

/**
 * Accept terms of service
 */
export async function acceptTerms() {
  const response = await authFetch('/api/auth/accept-terms', {
    method: 'POST',
  });

  return { ok: response.ok, data: await response.json() };
}

/**
 * Refresh access token
 */
export async function refreshAccessToken() {
  const { refreshToken } = getStoredTokens();

  if (!refreshToken) return false;

  try {
    const response = await fetch(`${API_BASE}/api/auth/refresh`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ refreshToken }),
    });

    if (response.ok) {
      const data = await response.json();
      storeTokens(data.tokens.accessToken, null);
      return true;
    }
  } catch (error) {
    console.error('[AUTH] Refresh failed:', error);
  }

  return false;
}

/**
 * Get session status
 */
export async function getSessionStatus() {
  const response = await authFetch('/api/auth/session-status');
  return { ok: response.ok, data: await response.json() };
}

/**
 * Get current user
 */
export async function getCurrentUser() {
  const response = await authFetch('/api/auth/me');
  return { ok: response.ok, data: await response.json() };
}

/**
 * Logout
 */
export async function logout() {
  const { refreshToken } = getStoredTokens();

  try {
    await authFetch('/api/auth/logout', {
      method: 'POST',
      body: JSON.stringify({ refreshToken }),
    });
  } catch (error) {
    console.error('[AUTH] Logout error:', error);
  }

  clearTokens();
}

/**
 * Send heartbeat to update activity
 */
export async function heartbeat() {
  const response = await authFetch('/api/auth/heartbeat', {
    method: 'POST',
  });

  return { ok: response.ok, data: await response.json() };
}

/**
 * Change password
 */
export async function changePassword(currentPassword, newPassword) {
  const response = await authFetch('/api/auth/change-password', {
    method: 'POST',
    body: JSON.stringify({ currentPassword, newPassword }),
  });

  return { ok: response.ok, data: await response.json() };
}

/**
 * Verify password (for secondary authentication)
 */
export async function verifyPassword(password) {
  const response = await authFetch('/api/auth/verify-password', {
    method: 'POST',
    body: JSON.stringify({ password }),
  });

  return { ok: response.ok, data: await response.json() };
}

/**
 * Request password reset
 */
export async function requestPasswordReset(email) {
  const response = await fetch(`${API_BASE}/api/auth/request-reset`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ email }),
  });

  return { ok: response.ok, data: await response.json() };
}

/**
 * Reset password with code
 */
export async function resetPassword(email, code, newPassword) {
  const response = await fetch(`${API_BASE}/api/auth/reset-password`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ email, code, newPassword }),
  });

  return { ok: response.ok, data: await response.json() };
}

/**
 * Get portal URL for a role
 */
export function getPortalUrl(role) {
  const isDev = import.meta.env.DEV;

  // Development URLs (localhost)
  const devPortalMap = {
    holdings_admin: 'http://localhost:5174',
    franchise_owner: 'http://localhost:5175',
    manager: 'http://localhost:5175',
    driver: 'http://localhost:5176',
    customer: 'http://localhost:5177',
    employer: 'http://localhost:5177',
    employee: 'http://localhost:5177',
    investor: 'http://localhost:5177',
    sba_lender: 'http://localhost:5177',
  };

  // Production URLs
  const prodPortalMap = {
    holdings_admin: 'https://hq.stafull.com',
    franchise_owner: 'https://app.stafull.com',
    manager: 'https://app.stafull.com',
    driver: 'https://driver.stafull.com',
    customer: 'https://my.stafull.com',
    employer: 'https://my.stafull.com',
    employee: 'https://my.stafull.com',
    investor: 'https://my.stafull.com',
    sba_lender: 'https://my.stafull.com',
  };

  const portalMap = isDev ? devPortalMap : prodPortalMap;
  const defaultUrl = isDev ? 'http://localhost:5177' : 'https://my.stafull.com';

  return portalMap[role] || defaultUrl;
}

/**
 * Check territory availability by zip code
 */
export async function checkTerritoryAvailability(zipCode) {
  const response = await fetch(`${API_BASE}/api/territories/check-availability`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ zipCode }),
  });

  return { ok: response.ok, data: await response.json() };
}

/**
 * Register expansion interest (for areas without territory coverage)
 */
export async function registerExpansionInterest({ email, zipCode, city, state }) {
  const response = await fetch(`${API_BASE}/api/expansion-interest`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ email, zipCode, city, state }),
  });

  return { ok: response.ok, data: await response.json() };
}

export default {
  login,
  register,
  logout,
  sendVerification,
  verifyEmail,
  verifyPassword,
  acceptTerms,
  refreshAccessToken,
  getSessionStatus,
  getCurrentUser,
  heartbeat,
  changePassword,
  requestPasswordReset,
  resetPassword,
  getStoredTokens,
  storeTokens,
  clearTokens,
  getPortalUrl,
  checkTerritoryAvailability,
  registerExpansionInterest,
};
